### PR TITLE
[webgpu] Use workPerThread for fromPixels

### PR DESCRIPTION
**Pull request by axinging**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6530)

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @xhcao @haoyunfeix @gyagp PTAL

File: tfjs-backend-webgpu/src/from_pixels_webgpu.ts [hunk 1]
```diff
@@ -33,8 +34,14 @@ export class FromPixelsProgram implements WebGPUProgram {
   constructor(outputShape: number[], useImport = false) {
     this.outputShape = outputShape;
     this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    util.assert(
+        this.outputShape.length >= 1,
+        () => `outputShape.length of FromPixelsProgram is ${
+            this.outputShape.length}`);
+    this.workPerThread = this.outputShape[this.outputShape.length - 1];
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/from_pixels_webgpu.ts:41]: Is it better to directly pass `numChannels` to this constructor for understanding? So you can directly use this.workPerThread = numChannels or `this.dispatch = computeDispatch(
        this.dispatchLayout, this.outputShape, this.workGroupSize,
        [numChannels , 1, 1]);`

File: tfjs-backend-webgpu/src/from_pixels_webgpu.ts [hunk 2]
```diff
@@ -33,8 +34,14 @@ export class FromPixelsProgram implements WebGPUProgram {
...
     this.dispatch = computeDispatch(
-        this.dispatchLayout, this.outputShape, this.workGroupSize);
+        this.dispatchLayout, this.outputShape, this.workGroupSize,
+        [this.workPerThread, 1, 1]);
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/from_pixels_webgpu.ts:44]: Good catch!

File: tfjs-backend-webgpu/src/kernels/FromPixels.ts [hunk 3]
```diff
@@ -64,37 +62,40 @@ export function fromPixels(args: {
...
-    if (fromPixels2DContext == null) {
-      fromPixels2DContext = document.createElement('canvas').getContext('2d');
-    }
-    fromPixels2DContext.canvas.width = width;
-    fromPixels2DContext.canvas.height = height;
-    fromPixels2DContext.drawImage(
-        pixels as HTMLVideoElement | HTMLImageElement, 0, 0, width, height);
-    pixels = fromPixels2DContext.canvas;
-  }
+  if (useImport || useKernel) {
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/FromPixels.ts:86]: nit: ->`if (useKernel)`. I think `useImport` is true, `useKernel` must be also true.

**Review by qjia7 — APPROVED** [review]
LGTM with two nits.

File: tfjs-backend-webgpu/src/kernels/FromPixels.ts [hunk 4]
```diff
@@ -64,37 +62,40 @@ export function fromPixels(args: {
...
-      externalImage: pixels as HTMLCanvasElement | ImageBitmap,
-      backend,
-      attrs,
-      outShape,
-      useImport: false
-    });
+    const uniformData = [
+      {type: 'uint32', data: [size]}, {type: 'uint32', data: [numChannels]},
+      {type: 'uint32', data: [...strides]},
+      {type: 'uint32', data: [...program.dispatch]}
```
[the remark this answers is not in the document]
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/FromPixels.ts]: done

File: tfjs-backend-webgpu/src/kernels/FromPixels.ts (hunk 3 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/kernels/FromPixels.ts]: done

File: tfjs-backend-webgpu/src/from_pixels_webgpu.ts (hunk 1 above)
Comment (axinging) [reply tfjs-backend-webgpu/src/from_pixels_webgpu.ts]: done

**Pull request merged**